paderbox/io/new_subdir.py

```diff
@@ -0,0 +1,341 @@
+            raise ValueError(
+                f'Tried to download {name_type!r}.\nCould not open\n{url}\n'
+                'See in\n'
+                'https://github.com/andreasonny83/unique-names-generator/tree/master/src/dictionaries\n'
+                'for valid names.'
+            )
+        # https://stackoverflow.com/a/19156107/5766934
+        data = resource.read().decode(resource.headers.get_content_charset())
+
+        data = re.findall("'(.*?)'", data)
+
+        pb.io.dump(data, file)
+    else:
+        data = pb.io.load(file)
+    return data
+
+
+class NameGenerator:
+    """
+    >>> import numpy as np
+    >>> np.random.seed(0)
+    >>> ng = NameGenerator()
+    >>> ng()
+    'helpful_tomato_finch'
+    >>> ng.possibilities()  # With 28 million a collision is unlikely
+    28406196
+    >>> ng = NameGenerator(['adjectives', 'animals'])
+    >>> ng()
+    'colourful_tern'
+
+    """
+    def __init__(
+            self,
+            lists=('adjectives', 'colors', 'animals'),
+            separator='_',
+            rng=None,
+            replace=True,
+            # style='default',  # 'capital', 'upperCase', 'lowerCase'
+    ):
+        self.lists = [
+            _get_list_from_unique_names_generator(l)
+            if isinstance(l, str) else list(map(str, l))
+            for l in lists
+        ]
+        self.separator = separator
+        if rng is None:
+            import numpy as np
+            rng = np.random
+        self.rng = rng
+        self.replace = replace
+        self.seen = set()
+
+    def __call__(self):
+        for _ in range(1000):
+            name_parts = [
+                self.rng.choice(d)
+                for d in self.lists
+            ]
+            name = self.separator.join(name_parts)
+
+            if self.replace:
+                break
+            elif name in self.seen:
+                continue
+            else:
+                self.seen.add(name)
+                break
+        else:
+            raise RuntimeError("Couldn't find a new name.")
+
+        return name
+
+    def possibilities(self):
+        import numpy as np
+        return np.prod([[len(d) for d in self.lists]])
+
+
+if __name__ == '__main__':
+    def cli(
+            basedir,
+            *,
+            id_naming='index',
+            mkdir=True,
+            prefix=None,
+            suffix=None,
+            consider_mpi=False,  # useless for cli?
+            dry_run=False,
+    ):
+        """
+
+        Args:
+            basedir:
+            id_naming: e.g. 'index', 'time', 'adjective_color_animal'
+            mkdir:
+            prefix:
+            suffix:
+            consider_mpi:
+            dry_run:
+
+        Returns:
+
+        """
+        # Add more cli mappings if necessary
+        if id_naming == 'adjective_color_animal':
+            id_naming = NameGenerator(('adjectives', 'colors', 'animals'))
```

Comment:
Put this into `get_new_subdir`?

Reply:
I am not sure, if I should put this inside (Actually I was unsure if the cli should have this).
On the python side using `NameGenerator(('adjectives', 'colors', 'animals'))` is more verbose and it is obvious, hot to change the generation.

Reply:
okay